### PR TITLE
Exclude CoreTelephony from MacCatalyst build

### PR DESCRIPTION
**Pull request by DmitriyKirakosyan**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* ~[ ] Did you add unit tests?~
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR fixes the build failure when building for macCatalyst with Swift 5.5:
```
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_CTTelephonyNetworkInfo", referenced from:
      objc-class-ref in AppCenter.o
ld: symbol(s) not found for architecture x86_64
```

This error breaks a SPM integration in Xcode 13 for Mac Catalyst platform.

## Related PRs or issues

[AB#88386](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/88386)
#2359

### CODE REVIEW

File: Package@swift-5.3.swift [hunk 1]
```diff
@@ -47,7 +47,7 @@ let package = Package(
                 .linkedFramework("SystemConfiguration"),
                 .linkedFramework("AppKit", .when(platforms: [.macOS])),
                 .linkedFramework("UIKit", .when(platforms: [.iOS, .tvOS])),
-                .linkedFramework("CoreTelephony", .when(platforms: [.iOS, .macOS])),
+                .linkedFramework("CoreTelephony", .when(platforms: [.iOS])),
```
Comment (AnastasiaKubova) [inline Package@swift-5.3.swift:50]: Please update the changelog and fill the description template.

Comment (DmitriyKirakosyan) [reply Package@swift-5.3.swift:50]: cc @aleksandr-dorofeev

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 4.3.1 (Under development)
 
+### App Center 
+
+* **[Fix]** Remove CoreTelephony dependecy for maCatalyst platform. This fixed a broken SPM integration for macCatalyst when using swift 5.5
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:7]: Fix `Undefined symbol: OBJC_CLASS$_CTTelephonyNetworkInfo` error for Mac Catalyst platform when integrating the SDK via Swift Package Manager with Swift 5.5 and higher.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 4.3.1 (Under development)
 
+### App Center 
+
+* **[Fix]** Fix Undefined symbol: OBJC_CLASS$_CTTelephonyNetworkInfo error for Mac Catalyst platform when integrating the SDK via Swift Package Manager with Swift 5.5 and higher.
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:7]: `Undefined symbol: OBJC_CLASS$_CTTelephonyNetworkInfo`

**Review by AnastasiaKubova — APPROVED** [review]

**Pull request merged**